onboarding_auth/onboardingAuthPresentation/src/main/java/com/alexafit/onboardingAuthPresentation/login/LoginScreen.kt

```diff
@@ -0,0 +1,157 @@
+import com.alexafit.core.util.UiEvent
+import com.alexafit.coreui.LightBlue
+import com.alexafit.coreui.LocalSpacing
+import com.alexafit.coreui.components.buttons.TextActionButton
+import com.alexafit.coreui.components.textfield.TextField
+import com.alexafit.onboardingAuthPresentation.R
+import com.alexafit.onboardingAuthPresentation.login.event.LoginEvent
+
+@Composable
+fun LoginScreen(
+    scaffoldState: ScaffoldState,
+    onEventClick: (LoginEvent) -> Unit,
+    viewModel: LoginViewModel = hiltViewModel()
+) {
+    val spacing = LocalSpacing.current
+    val context = LocalContext.current
+    LaunchedEffect(key1 = true) {
+        viewModel.uiEvent.collect { event ->
+            when (event) {
+                is UiEvent.Success -> onEventClick(LoginEvent.Login)
+                is UiEvent.ShowSnackbar -> {
+                    scaffoldState.snackbarHostState.showSnackbar(
+                        message = event.message.asString(context)
+                    )
+                }
+            }
+        }
+    }
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colors.background)
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                text = stringResource(id = R.string.title_welcome_back),
+                textAlign = TextAlign.Center,
+                style = MaterialTheme.typography.h1,
+                color = MaterialTheme.colors.onBackground,
+                modifier = Modifier
+                    .padding(vertical = spacing.spaceLarge)
+            )
+            Spacer(modifier = Modifier.height(spacing.spaceMedium))
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(
+                        MaterialTheme.colors.surface,
+                        shape = RoundedCornerShape(
+                            topStart = spacing.spaceLarge,
+                            topEnd = spacing.spaceLarge
+                        )
+                    )
+                    .padding(
+                        vertical = spacing.spaceLarge,
+                        horizontal = spacing.spaceSmall
+                    ),
+                verticalArrangement = Arrangement.Top,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                // Need to finish TextField Logic and responsive design
+                Spacer(modifier = Modifier.height(spacing.spaceSmall))
+                TextField(
+                    text = "",
+                    hint = stringResource(id = R.string.text_field_hint_email),
+                    keyboardImeAction = ImeAction.Next,
+                    iconVector = null,
+                    iconVectorDescription = null,
```

Comment:
To be a bit more flexible, you could also pass a composable lambda here to leave it up to the caller what they want at the end of the text field to not force them to use an icon

Reply:
I will do that then. Sounds like a good idea